reporting/src/main/java/org/wso2/testgrid/reporting/EmailReportProcessor.java

```diff
@@ -69,76 +70,73 @@ public EmailReportProcessor() {
      * @param product product needing the results
      * @return list of test-plan sections
      */
-    public List<TestPlanResultSection> generatePerTestPlanSection(Product product, List<TestPlan> testPlans)
+    public List<TPResultSection> generatePerTestPlanSection(Product product, List<TestPlan> testPlans)
             throws ReportingException {
-        List<TestPlanResultSection> perTestPlanMap = new ArrayList<>();
+        List<TPResultSection> perTestPlanList = new ArrayList<>();
         String testGridHost = ConfigurationContext.getProperty(ConfigurationContext.
                 ConfigurationProperties.TESTGRID_HOST);
         String productName = product.getName();
         for (TestPlan testPlan : testPlans) {
             if (testPlan.getStatus().equals(Status.SUCCESS)) {
+                logger.debug(String.format("Test plan ,%s, status is set to success.Not including in email report. "
+                        + "Infra combination: %s", testPlan.getId(), testPlan.getInfraParameters()));
                 continue;
             }
             String deploymentPattern = testPlan.getDeploymentPattern().getName();
             String testPlanId = testPlan.getId();
-            TestPlanResultSection testPlanResultSection = new TestPlanResultSection();
-            testPlanResultSection.setDeployment(deploymentPattern);
-            testPlanResultSection.setInfraCombination(testPlan.getInfrastructureConfig().getParameters().toString());
-            try {
-                testPlanResultSection.setResult(getIntegrationTestLogResult(testPlan));
-            } catch (ReportingException e) {
-                logger.error("Integration test results of the test-plan with id " + testPlan.getId() +
-                        " is not added to the email-report due to a reporting exception occurred.", e.getMessage());
+            final String infraCombination = testPlan.getInfraParameters();
+            final String dashboardLink = String.join("/", testGridHost, productName, deploymentPattern,
+                    TEST_PLANS_URI, testPlanId);
+
+            SurefireReporter surefireReporter = new SurefireReporter();
```

Comment:
We can take the SurefireReporter out of the loop of test-plans and avoid creating objects per-loop-turn